Data/HashMap/Base.hs

```diff
@@ -401,20 +419,18 @@ instance H.Hashable2 HashMap where
           = s `hashLeafWithSalt` l `go` tl
         -- For collisions we hashmix hash value
         -- and then array of values' hashes sorted
-        go s (Collision h a : tl)
-          = (s `H.hashWithSalt` h) `hashCollisionWithSalt` a `go` tl
+        go s (Collision _ l1 l2 hm' : tl)
+          = collisionHash l1 l2 hm' s `go` tl
         go s (_ : tl) = s `go` tl
 
+        collisionHash (L k1 v1) (L k2 v2) hm' s
+                = L.foldl' H.hashWithSalt s
+                $ L.sort . L.map (\(k, v) -> (s `hk` k) `hv` v)
+                $ (k1, v1) : (k2, v2) : toList hm'
```

Comment:
Why do we ignore the hash in the `Collision` node? We used to use it? Perhaps that makes sense for performance, but a comment would be in order.

Reply:
@treeowl Because we also ignore the hash incase of the `Leaf` constructor. I'm not sure why though. What would you like me to do here?

Reply:
I think it's okay for now. The whole thing is a bit weird, but it's not your fault.